src/driver/sap/SapQueryRunner.ts

```diff
@@ -237,7 +239,9 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
                });
            } catch (err) {
                 fail(err);
-            }
+           } finally {
+               statement?.drop();
```

Comment:
This might execute before callback where data is read is being called, since exec is not awaited since its callback based. I would put drop in the callback

Reply:
Good thinking. :)

Reply:
I've reworked it a little bit.  Let me know what you think!  I tried to reduce code duplication.

To be fair... this entire query method needs to be rewritten, it's a bit difficult to work with...